### PR TITLE
UI/vault 6212/multiple issuer pki changes

### PR DESCRIPTION
**Pull request by hellobontempo**
### UI updates to accommodate backend changes in #15277 
- copy changes from "Replace CA" to "Add" 
- disabling delete (for 1.11 all delete capabilities will only be available via CLI and API) 
<hr>
Screen Shot 2022-05-17 at 2 20 17 PM
Screen Shot 2022-05-17 at 2 20 42 PM

### CODE REVIEW

File: ui/app/components/config-pki-ca.js [hunk 1]
```diff
@@ -140,15 +140,10 @@ export default Component.extend({
       const isUpload = this.model.uploadPemBundle;
       model
         .save({ adapterOptions: { method } })
-        .then((m) => {
+        .then(() => {
           if (method === 'setSignedIntermediate' || isUpload) {
             this.send('refresh');
             this.flashMessages.success('The certificate for this backend has been updated.');
-          } else if (!m.get('certificate') && !m.get('csr')) {
```
Comment (hellobontempo) [inline ui/app/components/config-pki-ca.js:147]: According to the backend team, a certificate will always be generated so no need for this `else if`

File: ui/app/components/config-pki-ca.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ export default Component.extend({
    * @param DS.Model
    * @public
    *
-   * a `pki-config` model - passed in in the component useage
```
Comment (Monkeychip) [inline ui/app/components/config-pki-ca.js:55]: I also found this spelling in the config-pki.js row 23. Nice clean up work. Thank you.

File: ui/app/styles/components/popup-menu.scss [hunk 3]
```diff
@@ -8,7 +8,7 @@
     box-shadow: $box-shadow, $box-shadow-middle;
     padding: 0;
     position: relative;
-    width: 200px;
+    width: 210px;
```
Comment (Monkeychip) [inline ui/app/styles/components/popup-menu.scss:11]: This component is used a lot. Just want to make sure it's okay to change this globally.

File: ui/app/templates/components/config-pki-ca.hbs [hunk 4]
```diff
@@ -252,5 +237,24 @@
...
+      <div class="control">
+        <ToolTip @verticalPosition="above" @horizontalPosition="center" as |T|>
+          <T.Trigger data-test-tooltip-trigger tabindex="-1">
+            <button type="button" class="button is-primary" disabled={{true}}>
+              Delete
+            </button>
+          </T.Trigger>
+          <T.Content @defaultClass="tool-tip smaller-font">
+            <div class="box" data-test-hover-copy-tooltip-text>
+              Deleting a CA is only available via the CLI and API.
```
Comment (Monkeychip) [inline ui/app/templates/components/config-pki-ca.hbs:250]: this is honestly going to be a nice change for us. At least I think it is.

**Review by Monkeychip — APPROVED** [review]
Nice work!  A couple of non-blocking comments.

File: ui/app/components/config-pki-ca.js (hunk 1 above)
Comment (cipherboy) [reply ui/app/components/config-pki-ca.js:147]: Obviously, pending any actual errors. :-) But if its a valid request, you will always get a cert or CSR by design now.

File: ui/app/templates/components/config-pki-ca.hbs [hunk 5]
```diff
@@ -4,7 +4,11 @@
     {{#if this.needsConfig}}
       Configure CA Certificate
     {{else}}
-      Replace CA Certificate
+      {{#if this.model.certificate}}
+        Generated Certificate
```
Comment (hashishaw) [inline ui/app/templates/components/config-pki-ca.hbs:8]: Should this be `Generate` as in, the verb?

Comment (hellobontempo) [reply ui/app/templates/components/config-pki-ca.hbs:8]: This title shows when the certificate has been generated. Here's a screenshot from binary - so it shows the old `Replace` text. But this is what's been updated to say `Generated`

Screen Shot 2022-05-18 at 9 38 09 AM

Comment (cipherboy) [reply ui/app/templates/components/config-pki-ca.hbs:8]: It shows up here:

image

And I think generated is correct :-)

**Review by cipherboy — APPROVED** [review]
Looks good to me, thanks @hellobontempo!

File: ui/app/styles/components/popup-menu.scss (hunk 3 above)
Comment (hellobontempo) [reply ui/app/styles/components/popup-menu.scss:11]: I spot checked a couple other pop-up menus and it didn't seem to be an issue. But worth noting!

**Pull request merged**